epochStart/bootstrap/process.go

```diff
@@ -285,6 +268,21 @@ func (e *epochStartBootstrap) Bootstrap() (Parameters, error) {
 	return e.requestAndProcessing()
 }
 
+func (e *epochStartBootstrap) computeIfCurrentEpochIsSaved() bool {
+	e.initializeFromLocalStorage()
```

Comment:
We have a bug in the method initializeFromLocalStorage that can appear if we have an epoch DB saved but on the wrong shard. So, the method `getParentDirAndLastEpoch` should take into account also the shard ID

Reply:
@bogdan-rosianu is working on refactoring openStorage.go file .

Reply:
great!